src/utils/dayViewLayout.js

```diff
@@ -0,0 +1,237 @@
+ * groups of siblings, will be found. Both are needed in order to calculate the
+ * width of the first event. When the width is known, its siblings will be
+ * given the same width, but with an incremental x-offet.
+ *
+ * Each group of children will be looking to move as far away from its original
+ * parent as possible. A move can be made to one of the parent's siblings, if
+ * that sibling is also a parent to the child group. This will make room for
+ * more events.
+ *
+ * When a child group knows its parent, it looks at the space occupied by that
+ * parent, and calculates the remaning available space and divides that among
+ * each other.
+ *
+ * All widths and x-offsets are calculated without taking overlapping into
+ * account. Overlapping is added in the end according to the OVERLAP_MULTIPLIER.
+ * If that is set to 0, the events won't overlap or grow.
+ *
+ * When one of these rounds are finished, all events connected have been
+ * traversed, so the cursor will be moved past all of them.
+ */
+export default function getStyledEvents ({
+  events: unsortedEvents, startAccessor, endAccessor, min, totalMin, step
+}) {
+  let OVERLAP_MULTIPLIER = 0.3
+  let events = sort(unsortedEvents, { startAccessor, endAccessor })
+  let helperArgs = { events, startAccessor, endAccessor, min, totalMin, step }
+
+  let idx = 0
+
+  // One iteration will cover all connected events.
+  while (idx < events.length) {
+    let siblings = getSiblings(idx, helperArgs)
+    let { childGroups, nbrOfChildColumns } = getChildGroups(
+      idx, idx + siblings.length + 1, helperArgs
+    )
+    let nbrOfColumns = Math.max(nbrOfChildColumns, siblings.length) + 1;
+
+    // Set styles to top level events.
+    [idx, ...siblings].forEach((eventIdx, siblingIdx) => {
+      let width = 100 / nbrOfColumns
+      let xAdjustment = width * (nbrOfColumns > 1 ? OVERLAP_MULTIPLIER : 0)
+      let { top, height } = getYStyles(eventIdx, helperArgs)
+
+      events[eventIdx].styles = {
+        top,
+        height,
+        width: width + xAdjustment,
+        xOffset: (width * siblingIdx) - xAdjustment
+      }
+    })
+
+    childGroups.forEach(group => {
+      let parentIdx = idx
+      let siblingIdx = 0
+
+      // Move child group to sibling if possible, since this will makes
+      // room for more events.
+      while (isChild(siblings[siblingIdx], group[0], helperArgs)) {
+        parentIdx = siblings[siblingIdx]
+        siblingIdx++
+      }
+
+      // Set styles to child events.
+      group.forEach((eventIdx, i) => {
+        let parentStyles = events[parentIdx].styles
+        let spaceOccupiedByParent = parentStyles.width + parentStyles.xOffset
+        let columns = Math.min(group.length, nbrOfColumns)
+        let width = (100 - spaceOccupiedByParent) / columns
+        let xAdjustment = spaceOccupiedByParent * OVERLAP_MULTIPLIER
+        let { top, height } = getYStyles(eventIdx, helperArgs)
+
+        events[eventIdx].styles = {
```

Comment:
is this the user provided event object? I'd prefer not to mutate data if possible, maybe we can do `{ event, style }`?